pkg/operator/context/aggregators.go

```diff
@@ -23,31 +23,55 @@ import (
 	"github.com/cortexlabs/cortex/pkg/consts"
 	"github.com/cortexlabs/cortex/pkg/lib/errors"
 	"github.com/cortexlabs/cortex/pkg/lib/hash"
+	s "github.com/cortexlabs/cortex/pkg/lib/strings"
 	"github.com/cortexlabs/cortex/pkg/operator/api/context"
 	"github.com/cortexlabs/cortex/pkg/operator/api/resource"
 	"github.com/cortexlabs/cortex/pkg/operator/api/userconfig"
 	"github.com/cortexlabs/cortex/pkg/operator/config"
 )
 
 func loadUserAggregators(
-	aggregatorConfigs userconfig.Aggregators,
+	config *userconfig.Config,
 	impls map[string][]byte,
 	pythonPackages context.PythonPackages,
 ) (map[string]*context.Aggregator, error) {
 
 	userAggregators := make(map[string]*context.Aggregator)
-	for _, aggregatorConfig := range aggregatorConfigs {
+	for _, aggregatorConfig := range config.Aggregators {
 		impl, ok := impls[aggregatorConfig.Path]
 		if !ok {
 			return nil, errors.Wrap(ErrorImplDoesNotExist(aggregatorConfig.Path), userconfig.Identify(aggregatorConfig))
 		}
-		aggregator, err := newAggregator(*aggregatorConfig, impl, nil, pythonPackages)
+		aggregator, err := newAggregator(*aggregatorConfig, impl, nil, pythonPackages, false)
 		if err != nil {
 			return nil, err
 		}
 		userAggregators[aggregator.Name] = aggregator
 	}
 
+	for _, aggregateConfig := range config.Aggregates {
+		if aggregateConfig.AggregatorPath == nil {
+			continue
+		}
+
+		impl, ok := impls[*aggregateConfig.AggregatorPath]
+		if !ok {
+			return nil, errors.Wrap(ErrorImplDoesNotExist(*aggregateConfig.AggregatorPath), userconfig.Identify(aggregateConfig))
+		}
+
+		anonAggregatorConfig := &userconfig.Aggregator{
+			ResourceFields: userconfig.ResourceFields{
+				Name: s.PathToName(*aggregateConfig.AggregatorPath),
+			},
+			Path: *aggregateConfig.AggregatorPath,
+		}
+		aggregator, err := newAggregator(*anonAggregatorConfig, impl, nil, pythonPackages, true)
+		if err != nil {
+			return nil, err
+		}
+		userAggregators[*aggregateConfig.AggregatorPath] = aggregator
+	}
+
```

Comment:
If we set the `Aggregator` field of the `aggregate` to the autogenerated name, and used `userAggregators[s.PathToName(*aggregateConfig.AggregatorPath)]` instead of `userAggregators[*aggregateConfig.AggregatorPath]` would it make things easier down the line? E.g. I think we could remove the logic in `autogenerator.go` and lines 162-171 in this file. (same applies for transformers)